src/nn/Modules/Activations.cpp

```diff
@@ -46,5 +46,48 @@ namespace af
         {
             return max(input, m_slope * input);
         }
+
+        PReLU::PReLU(int size, double spread)
+        {
+            auto w = nn::weight(size, 1, spread);
+            setParams({w});
+        }
+
+        PReLU::PReLU(const Variable &w) :
+            Module({w})
+        {
+        }
+
+        Variable PReLU::forward(const Variable &input)
+        {
+            auto tmp = max(input, 0.0);
+            auto res = expandAs(m_parameters[0],tmp) * tmp;
+            //TODO: Determine if doing the max after the mul is preferable
+            return res;
+
+        }
+
+        ELU::ELU(double alpha) :
+            m_alpha(alpha)
+        {
+        }
+
+        Variable ELU::forward(const Variable &input)
+        {
+            auto res = max(input, m_alpha * (exp(input) - 1));
+            return res;
+        }
+
+        ThresholdedReLU::ThresholdedReLU(double threshold) :
+            m_threshold(threshold)
+        {
+        }
+
+        Variable ThresholdedReLU::forward(const Variable &input)
+        {
+            auto mask = input > m_threshold;
+            return input * mask;
+        }
```

Comment:
I believe `Threshold` may be misleading, since threshold usually denotes a function R -> {0,1} (e.g., `return input > m_threshold;`). `ThresholdedReLU` or `ThresholdReLU` seem a better fit to me.